embed/src/app.py

```diff
@@ -171,62 +161,24 @@ def embed_sentences(model: str, sentences: list[str]):
     return output
 
 
-def poll_queue_every(queue_uri: str, queue_name: str, interval: int = 5):
-    logging.debug(f"Starting queue poller for {queue_name} every {interval} seconds")
-    while True:
-        logging.debug(f"Polling queue {queue_name} (every {interval} seconds)")
-        poll_queue(queue_uri, queue_name)
-        time.sleep(interval)
-
-
-def poll_queue(queue_uri: str, queue_name: str, max_connection_attempts: int = 10):
-    logging.info(f"Polling queue {queue_name}")
-
-    connection = None
-
-    for i in range(max_connection_attempts):
-        try:
-            logging.debug(f"Attempting to connect to queue {queue_name} (attempt {i}/{max_connection_attempts})")
-            connection = pika.BlockingConnection(
-                parameters=pika.URLParameters(queue_uri),
-            )
-            logging.debug(f"Connected to queue {queue_name}")
-            break
-        except Exception as e:
-            logging.error(f"Failed to connect to queue attempt {i}: {e}")
-            time.sleep(5)
-
-    if not connection:
-        logging.error("Failed to connect to queue, shutting down")
-        return
-
+def subscribe_to_queue():
+    connection = env.blocking_connection()
     channel = connection.channel()
-    channel.queue_declare(queue=queue_name, durable=True)
-    channel.basic_consume(queue=queue_name, on_message_callback=embed_item_callback, auto_ack=False)
+    channel.queue_declare(queue=env.embed_queue_name, durable=True)
+    channel.basic_consume(queue=env.embed_queue_name, on_message_callback=embed_item_callback, auto_ack=False)
     channel.start_consuming()
 
 
-def embed_item_callback(ch, method, properties, body):
+def embed_item_callback(ch: BlockingChannel, method, properties, body):
     logging.info(f"Received message {method.delivery_tag} by callback")
-
     try:
         body_dict = json.loads(body.decode("utf-8"))
+        embed_queue_item = EmbedQueueItem(**body_dict)
+        response = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
+        logging.info(f"Embedding ID {response['embedding_id']} complete for {method.delivery_tag}")
     except json.JSONDecodeError as e:
         logging.error(f"Failed to decode message: {e}")
-        ch.basic_nack(delivery_tag=method.delivery_tag)
-        return
-    try:
-        embed_queue_item = EmbedQueueItem(**body_dict)
     except pydantic.ValidationError as e:
         logging.error(f"Failed to validate message: {e}")
-        ch.basic_nack(delivery_tag=method.delivery_tag)
-        return
-
-    output = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
-    output = EmbeddingResponse(**output)
-
-    # TODO: Send the output to Elasticsearch?
-
-    logging.info(f"Embedding ID {output.embedding_id} complete for {method.delivery_tag}")
-    ch.basic_ack(delivery_tag=method.delivery_tag)
-    ch.stop_consuming()
+    finally:
+        ch.basic_ack(delivery_tag=method.delivery_tag)
```

Comment:
if this is not `basic_nack` then the unprocessed items remain in the queue.... tbh i wasnt sure what we want to do here?